tests/end-to-end.rs

```diff
@@ -22,86 +41,366 @@ macro_rules! test_case {
+    let walkdir = WalkDir::new(format!("{test_working_dir}/cdk.out/"));
+
+    for entry in walkdir.into_iter().map(|e| e.unwrap()) {
+        let filename = entry.file_name().to_str().unwrap();
+        if filename.contains("template.json") {
+            println!("Comparing {filename} to the original template");
+            let res = std::process::Command::new("git")
+                .args([
+                    "diff",
+                    "--no-index",
+                    &format!("./tests/end-to-end/{test_name}/template.json"),
+                    entry.path().to_str().unwrap(),
+                ])
+                .output()
+                .expect("git diff failed");
+
+            let stack_name = filename.split(".").next().unwrap();
+            let mut f = fs::File::create(format!("{test_working_dir}/{stack_name}.diff")).unwrap();
+            f.write_all(&res.stdout).unwrap();
         }
     }
 }
 
-impl Drop for UpdateSnapshot<'_> {
-    fn drop(&mut self) {
-        use std::fs::File;
-        use std::io::Write;
-        use std::path::PathBuf;
-
-        if std::env::var_os("UPDATE_SNAPSHOTS").is_some() && self.actual != self.expected {
-            let path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
-                .join("tests")
-                .join(self.path);
-            let mut file = File::create(path).unwrap();
-            file.write_all(self.actual.as_bytes()).unwrap();
+fn update_snapshots(
+    cdk_stack_filename: &str,
+    cdk_app_filename: &str,
+    test_working_dir: &str,
+    expected_outputs_dir_name: &str,
+    snapshots_zip: &mut ZipArchive<Cursor<&[u8]>>,
+) {
+    if std::env::var_os("UPDATE_SNAPSHOTS").is_none() {
+        // By default, and in CI/CD, skip updating the snapshots
+        println!("Not updating snapshots because UPDATE_SNAPSHOTS is none.");
+        return;
+    }
+    println!("Updating snapshots...");
+
+    let expected_outputs_path = &format!("tests/end-to-end/{expected_outputs_dir_name}/");
+    create_dir_all(expected_outputs_path).unwrap();
+
+    // App file
+    // If the app file does not already exist in the snapshot, copy the one generated by this test run to the test's directory
+    if snapshots_zip.by_name(&format!("{expected_outputs_dir_name}/{cdk_app_filename}")).is_err() {
+        copy(
+            format!("{test_working_dir}/{cdk_app_filename}"),
+            format!("{expected_outputs_path}/{cdk_app_filename}"),
+        )
+        .unwrap();
+    }
+
+    // Stack file
+    copy(
+        format!("{test_working_dir}/{cdk_stack_filename}"),
+        format!("{expected_outputs_path}/{cdk_stack_filename}"),
+    )
+    .unwrap();
+
+    // Template and diff files
+    let walkdir = WalkDir::new(test_working_dir);
+    for entry in walkdir.into_iter().map(|e| e.unwrap()) {
+        let filename = entry.file_name().to_str().unwrap();
+        if entry.path().to_str().unwrap().contains("node_modules") {
+            continue;
+        }
```

Comment:
Our snaps can be non-JS/TS, right? We should also exclude any folders created by other languages for dependencies

Reply:
yes, that is coming very soon!